tutorials/mongoose/mongoose-validate-unique-email.md

```diff
@@ -0,0 +1,7 @@
+With Mongoose, you can prevent duplicates in your databases using [validation](https://mongoosejs.com/docs/validation.html). Validation is defined in the SchemaType and is a middleware. You can also create your own validation in the schema or you can use Mongooses's built in validation. To prevent duplicates, we recommend using the `unique` property as it tells Mongoose each document should have a [unique](https://masteringjs.io/tutorials/mongoose/unique) value for a given path. It is a shorthand for creating a MongoDB unique index on, in this case, `email`. If you wait for the index to be built, you can use either Mongoose's `Model.on('index')` event or you can you Mongoose's promised based alternative, `Model.init()`, which is shown below::
```

Comment:
I'd recommend getting rid of the `Model.on('index')` part, `await Model.init()` is much better